### PR TITLE
Delete `hetero` and `multi` configs + `ie_exception.hpp`

### PR DESCRIPTION
**Pull request by vurusovs**
CVS-130757

### CODE REVIEW

File: src/inference/src/compilation_context.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #endif
 
 #include "cpp/ie_cnn_network.h"
-#include "details/ie_exception.hpp"
+// #include "ie_common.h"
```
Comment (ilya-lavrenov) [inline src/inference/src/compilation_context.cpp:15]: please, remove commented code

Comment (vurusovs) [reply src/inference/src/compilation_context.cpp]: Done, thanks!

**Review by praasz — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Moving to draft as it depends on NPU

**Comment by p-durandin** [comment]
Blocked by https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/8698/

**Comment by vurusovs** [comment]
Waiting for https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/8698

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**